app/models/podcast_planner.rb

```diff
@@ -0,0 +1,152 @@
+class PodcastPlanner
```

Comment:
this is a new model that i'm using to package all the date calculation and draft generation logic, in order to abstract it away from the controller.